src/lib/y2network/widgets/boot_protocol.rb

```diff
@@ -0,0 +1,364 @@
+          # and future means when type cannot be changed and when cwm object tabs are used, as it has limited lifetime of cwm definition
+          _(
+            "<p>Check <b>iBFT</b> if you want to keep the network configured in your BIOS.</p>\n"
+          ) +
+          # Address dialog help 2/8
+          _(
+            "<p>Select <b>Dynamic Address</b> if you do not have a static IP address \nassigned by the system administrator or your Internet provider.</p>\n"
+          ) +
+          # Address dialog help 3/8
+          _(
+            "<p>Choose one of the dynamic address assignment methods. Select <b>DHCP</b>\n" \
+              "if you have a DHCP server running on your local network. Network addresses \n" \
+              "are then automatically obtained from the server.</p>\n"
+          ) +
+          # Address dialog help 4/8
+          _(
+            "<p>To search for an IP address and assign it statically, select \n" \
+              "<b>Zeroconf</b>. To use DHCP and fall back to zeroconf, select <b>DHCP + Zeroconf\n" \
+              "</b>. Otherwise, the network addresses must be assigned <b>Statically</b>.</p>\n"
+          )
+
+        if Yast::ProductFeatures.GetBooleanFeature("network", "force_static_ip")
+          res += _(
+            "<p>DHCP configuration is not recommended for this product.\nComponents of this product might not work with DHCP.</p>"
+          )
+        end
+
+        res
+      end
+
+      def dynamic_enabled(value)
+        Yast::UI.ChangeWidget(Id(:bootproto_dyn), :Enabled, value)
+        if value
+          # dhcp mode works only with plain dhcp
+          if :bootproto_dhcp == Yast::UI.QueryWidget(Id(:bootproto_dyn), :Value)
+            Yast::UI.ChangeWidget(Id(:bootproto_dhcp_mode), :Enabled, value)
+          else
+            Yast::UI.ChangeWidget(Id(:bootproto_dhcp_mode), :Enabled, false)
+          end
+        else
+          Yast::UI.ChangeWidget(Id(:bootproto_dhcp_mode), :Enabled, value)
+        end
+      end
+
+      def none_enabled(value)
+        Yast::UI.ChangeWidget(Id(:bootproto_ibft), :Enabled, value) if ibft_available?
+      end
+
+      def static_enabled(value)
+        Yast::UI.ChangeWidget(Id(:bootproto_ipaddr), :Enabled, value)
+        Yast::UI.ChangeWidget(Id(:bootproto_netmask), :Enabled, value)
+        Yast::UI.ChangeWidget(Id(:bootproto_hostname), :Enabled, value)
+      end
+
+      def value
+        Yast::UI.QueryWidget(Id(:bootproto), :CurrentButton)
+      end
+
+      def valid_netmask(ip, mask)
+        valid_mask = false
+        mask = mask[1..-1] if mask.start_with?("/")
+
+        if Yast::IP.Check4(ip) && (Yast::Netmask.Check4(mask) || Yast::Netmask.CheckPrefix4(mask))
+          valid_mask = true
+        elsif Yast::IP.Check6(ip) && Yast::Netmask.Check6(mask)
+          valid_mask = true
+        else
+          log.warn "IP address #{ip} is not valid"
```

Comment:
a bit strange when method is named ```valid_netmask``` ;-) May be it should be ```valid_static_ip_setup``` or something like that

Reply:
well, original name is validPrefixOrNetmask(ipa, mask) so I just try to shorten it :)